lib/publiccloud/instance.pm

```diff
@@ -919,9 +919,15 @@ sub do_systemd_analyze_time {
 
 sub upload_supportconfig_log {
     my ($self, %args) = @_;
-    $self->ssh_script_run(cmd => 'sudo supportconfig -R /var/tmp -B supportconfig -x AUDIT', timeout => 600);
-    $self->ssh_script_run(cmd => 'sudo chmod 0644 /var/tmp/scc_supportconfig.txz');
-    $self->upload_log('/var/tmp/scc_supportconfig.txz', failok => 1, timeout => 600);
+    my $timeout = 600;
+    my $start = time();
+    my $logs = "/var/tmp/scc_supportconfig";
+    my $cmd = "printf '\\n' | timeout $timeout sudo supportconfig -R " . dirname($logs) . " -B supportconfig -x AUDIT";
+    # bsc#1250310: to avoid some hang cases, inject '\n' in ssh supportconfig, as w.a.
```

Comment:
I suggest moving this comment one line above.

Reply:
done